blockchain/pending.go

```diff
@@ -44,6 +44,9 @@ func (p *PendingState) ContractNonce(addr *felt.Felt) (*felt.Felt, error) {
 		return nonce, nil
 	}
 
+	if p.isDeployedInPending(addr) {
+		return &felt.Zero, nil
```

Comment:
Is it risky that we use `&felt.Zero` in many places? It seems we might mutate it on accident.

Reply:
> mutate it on accident

Yeah, we should move away from using felt pointers as soon as possible.